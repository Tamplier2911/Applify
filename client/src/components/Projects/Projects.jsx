// import "./Projects.scss";
import React, { useState, useEffect } from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  setCurrentProjectFirst,
  setCurrentProjectSecond,
  setCurrentProjectThird,
  setCurrentProjectFourth,
  setCurrentProjectFifth,
  setCurrentProjectNext,
  getCurrentProject,
  // setCurrentProjectPrev
} from "../../redux/projects/projects.actions";
import { selectCurrentProject } from "../../redux/projects/projects.selectors";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
import { selectCurrentTheme } from "../../redux/theme/theme.selectors";

// projects constants
import projectsHeaders from "./ProjectsConstants";

// JS Rendering CSS
import {
  ProjectsContainer,
  ProjecstTitle,
  ProjectsCircles,
  ProjectsCircleZero,
  ProjectsCircleOne,
  ProjectsCircleTwo,
  ProjectsCircleThree,
  ProjectsCirclFour,
  ProjectsImageWrapper,
  ProjectsImage,
  ProjectsBox,
  ProjectsContent,
  ProjectsContentTitle,
  ProjectsContentDesc,
  ProjectsContentDescTitle,
  ProjectsContentDescText,
  ProjectsContentBottom,
  ProjectsContentDate,
  ProjectsContentLink,
} from "./ProjectsStyles";

const Projects = ({
  currentProject: { id, name, image, color, description, techStack, date, url },
  setCurrentProjectFirst,
  setCurrentProjectSecond,
  setCurrentProjectThird,
  setCurrentProjectFourth,
  setCurrentProjectFifth,
  setCurrentProjectNext,
  getCurrentProject,
  lang,
  theme,
  // setCurrentProjectPrev
}) => {
  const [isLoaded, setIsLoaded] = useState(false);
  useEffect(() => {
    // invoke this code only once to init current project
    if (!id) {
      getCurrentProject(lang);
    }

    if (id === 1) setCurrentProjectSecond(lang);
    if (id === 2) setCurrentProjectThird(lang);
    if (id === 3) setCurrentProjectFourth(lang);
    if (id === 4) setCurrentProjectFifth(lang);

    // animation state
    setIsLoaded(false);
    setTimeout(() => {
      setIsLoaded(true);
    }, 600);
    // 400 ?

    // slider functionality
    const slider = setInterval(() => {
      setCurrentProjectNext(lang);
    }, 20000);
    return () => clearInterval(slider);
  }, [
    setCurrentProjectNext,
    setCurrentProjectSecond,
    setCurrentProjectThird,
    setCurrentProjectFourth,
    setCurrentProjectFifth,
    getCurrentProject,
    lang,
    id,
  ]);

  const {
    projectsHeader,
    projectsDescription,
    projectsTechStack,
    projectsVisit,
  } = projectsHeaders[lang];

  date = date
    ? date.toLocaleString("en-us", {
        day: "numeric",
        month: "long",
        year: "numeric",
      })
    : undefined;

  return (
    <ProjectsContainer
      data-test="projects"
      animateIn={"bounceInRight"}
      animateOnce={true}
    >
      <ProjecstTitle>{projectsHeader}</ProjecstTitle>
      <ProjectsCircles>
        <ProjectsCircleZero
          data-test="projects-set-first"
          onClick={() => setCurrentProjectFirst(lang)}
          id={id}
        />
        <ProjectsCircleOne
          data-test="projects-set-second"
          onClick={() => setCurrentProjectSecond(lang)}
          id={id}
        />
        <ProjectsCircleTwo
          data-test="projects-set-third"
          onClick={() => setCurrentProjectThird(lang)}
          id={id}
        />
        <ProjectsCircleThree
          data-test="projects-set-fourth"
          onClick={() => setCurrentProjectFourth(lang)}
          id={id}
        />
        <ProjectsCirclFour
          data-test="projects-set-fifth"
          onClick={() => setCurrentProjectFifth(lang)}
          id={id}
        />
      </ProjectsCircles>
      <ProjectsImageWrapper load={isLoaded}>
        <ProjectsImage src={`${image}`} alt="Project presentation slide." />
      </ProjectsImageWrapper>
      <ProjectsBox data-test="projects-box">
        <ProjectsContent load={isLoaded}>
          <ProjectsContentTitle>{name}</ProjectsContentTitle>
          <ProjectsContentDesc>
            <ProjectsContentDescTitle color={color}>
              {projectsDescription}
            </ProjectsContentDescTitle>
            <ProjectsContentDescText>{description}</ProjectsContentDescText>
          </ProjectsContentDesc>
          <ProjectsContentDesc>
            <ProjectsContentDescTitle color={color}>
              {projectsTechStack}
            </ProjectsContentDescTitle>
            <ProjectsContentDescText>{techStack}</ProjectsContentDescText>
          </ProjectsContentDesc>
          <ProjectsContentBottom>
            <ProjectsContentDate>{date}</ProjectsContentDate>
            <ProjectsContentLink
              href={`${url}`}
              color={color}
              theme={theme}
              target="_blank"
              rel="noopener noreferrer"
            >
              {projectsVisit}
            </ProjectsContentLink>
          </ProjectsContentBottom>
        </ProjectsContent>
      </ProjectsBox>
    </ProjectsContainer>
  );
};

const mapStateToProps = () =>
  createStructuredSelector({
    currentProject: selectCurrentProject,
    lang: selectCurrentLanguage,
    theme: selectCurrentTheme,
  });

export default connect(mapStateToProps, {
  setCurrentProjectFirst,
  setCurrentProjectSecond,
  setCurrentProjectThird,
  setCurrentProjectFourth,
  setCurrentProjectFifth,
  setCurrentProjectNext,
  getCurrentProject,
  // setCurrentProjectPrev
})(Projects);
