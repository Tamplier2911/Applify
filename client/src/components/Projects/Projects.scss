.projects {
  display: grid;
  grid-template-columns: repeat(10, 1fr);
  grid-template-rows: repeat(10, min-content);

  padding: 1rem;

  &__title {
    grid-area: 1 / 1 / 2 / -1;
    font-family: var(--font-title);
    font-size: 4rem;
    font-weight: 300;

    padding: 1rem;
  }

  &__circles {
    grid-area: 2 / 6 / 3 / 9;
    display: grid;
    grid-template-columns: repeat(4, min-content);
    justify-content: end;
    grid-column-gap: 3rem;

    padding: 2rem;

    @media only screen and (max-width: 720px) {
      margin-top: 2rem;
      grid-area: 20 / 4 / 21 / 8;
      justify-content: center;
    }

    &--1 {
      cursor: pointer;
      width: 2rem;
      height: 2rem;
      background-color: var(--cl-dark-grey);
      background-color: var(--cl-dark-grey);

      border-radius: 50%;
      box-shadow: 0.1rem 0.1rem 0.4rem var(--cl-square-shadow);
      transition: background-color 0.3s;
      &:hover {
        background-color: var(--cl-darkest-grey);
      }
    }

    &--2 {
      cursor: pointer;
      width: 2rem;
      height: 2rem;
      background-color: var(--cl-dark-grey);

      border-radius: 50%;
      box-shadow: 0.1rem 0.1rem 0.4rem var(--cl-square-shadow);
      transition: background-color 0.3s;
      &:hover {
        background-color: var(--cl-darkest-grey);
      }
    }

    &--3 {
      cursor: pointer;
      width: 2rem;
      height: 2rem;
      background-color: var(--cl-dark-grey);

      border-radius: 50%;
      box-shadow: 0.1rem 0.1rem 0.4rem var(--cl-square-shadow);
      transition: background-color 0.3s;
      &:hover {
        background-color: var(--cl-darkest-grey);
      }
    }

    &--4 {
      cursor: pointer;
      width: 2rem;
      height: 2rem;
      background-color: var(--cl-dark-grey);

      border-radius: 50%;
      box-shadow: 0.1rem 0.1rem 0.4rem var(--cl-square-shadow);
      transition: background-color 0.3s;
      &:hover {
        background-color: var(--cl-darkest-grey);
      }
    }
  }

  &__imageContainer {
    grid-area: 2 / 2 / 8 / 5;
    // width: 20rem;
    // height: 20rem;

    box-shadow: 0 0.2rem 0.5rem var(--cl-square-shadow);
    animation: fadein 15s infinite;
    z-index: 5;

    @media only screen and (max-width: 720px) {
      grid-area: 2 / 2 / 10 / 10;
    }
  }

  &__image {
    width: 100%;
    height: 100%;
    display: block;
    object-fit: cover;
  }

  &__box {
    grid-area: 3 / 3 / 10 / 10;
    display: grid;
    grid-template-columns: 3fr 7.5fr;

    // height: 100rem;
    // width: 200%;
    background-color: var(--cl-grey);
    box-shadow: 0 0.2rem 1rem var(--cl-square-shadow);

    padding: 2rem;

    @media only screen and (max-width: 720px) {
      grid-area: 7 / 1 / 20 / 11;
      z-index: 10;
    }
  }

  &__content {
    // background-color: var(--cl-tiles-5);
    grid-column: 2 / 3;
    display: grid;
    grid-row-gap: 2rem;

    padding: 1rem;

    animation: fadein 15s infinite;

    @media only screen and (max-width: 720px) {
      grid-column: 1 / 3;
    }

    &--title {
      font-size: 2rem;
    }

    &--description {
      & span {
        font-weight: bold;
        color: var(--cl-tiles-4);
      }
    }

    &--bottom {
      display: grid;
      grid-template-columns: 1fr min-content;
      align-items: center;
    }

    &--date {
    }

    &--link {
      background-color: var(--cl-tiles-4);
      padding: 0.5rem 5rem;
      font-family: var(--font-title);
      color: var(--cl-white);
      text-decoration: none;

      transition: 0.3s background-color;

      &:hover {
        // color: var(--cl-font);
        background-color: var(--cl-tiles-2);
      }
    }
  }
}

.bullet-active-0 {
  background-color: var(--cl-black);
}

.bullet-active-1 {
  background-color: #55c57a;
}

.bullet-active-2 {
  background-color: #dd4f4f;
}

.bullet-active-3 {
  background-color: var(--cl-black);
}

// .text-active {

// }

@keyframes fadein {
  0% {
    opacity: 0;
  }
  20% {
    opacity: 1;
  }
  100% {
    opacity: 1;
  }
}
