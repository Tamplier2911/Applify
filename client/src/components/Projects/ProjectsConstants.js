const projectsHeaders = {
  eng: {
    projectsHeader: "Recent projects:",
    projectsDescription: "Description:",
    projectsTechStack: "Technology Stack:",
    projectsVisit: "Visit"
  },
  rus: {
    projectsHeader: "Недавние проекты:",
    projectsDescription: "Описание:",
    projectsTechStack: "Стек Технологий:",
    projectsVisit: "Посетить"
  },
  ukr: {
    projectsHeader: "Нещодавні проекти:",
    projectsDescription: "Опис:",
    projectsTechStack: "Стек Технологій:",
    projectsVisit: "Відвідати"
  }
};

export default projectsHeaders;
