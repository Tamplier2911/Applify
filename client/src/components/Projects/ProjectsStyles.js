import styled, { css } from "styled-components";
import ScrollAnimation from "react-animate-on-scroll";

/*

// before projects animation was faked using this mechanic:
// new projects pops in every 15 second via set interval
// this animation runs 15 second 0-20% fade in 20-100% resting state. 
const projectsAnimation = css`
  @keyframes fadein {
    0% {
      opacity: 0;
    }
    4% {
      opacity: 0;
    }
    20% {
      opacity: 1;
    }
    100% {
      opacity: 1;
    }
  }
`;

*/

const getThemeColorForLink = (props) => {
  const { theme } = props;
  if (theme === "dark") {
    return `color: var(--cl-font);`;
  } else {
    return `color: var(--cl-white);`;
  }
};

const fadeInProject = (props) => {
  const { load } = props;
  if (!load) {
    return `opacity: 0;`;
  } else if (load) {
    return `
        opacity: 1;
        transition: opacity 1s ease-in;
      `;
  }
};

const circlesSharedStyle = css`
  cursor: pointer;
  width: 2rem;
  height: 2rem;
  border-radius: 50%;
  transition: background-color 0.3s;
`;

const getLinkColor = (props) => {
  return props.color ? `${props.color}` : `var(--cl-tiles-2)`;
};

const getTitleColors = (props) => {
  return props.color ? `${props.color}` : `var(--cl-tiles-2)`;
};

const getCircleColorZero = (props) => {
  return props.id === 0 ? "var(--cl-circle-0)" : "var(--cl-circle-transit)";
};
const getCircleColorOne = (props) => {
  return props.id === 1 ? "var(--cl-circle-1)" : "var(--cl-circle-transit)";
};
const getCircleColorTwo = (props) => {
  return props.id === 2 ? "var(--cl-cirlce-2)" : "var(--cl-circle-transit)";
};
const getCircleColorThree = (props) => {
  return props.id === 3 ? "var(--cl-circle-3)" : "var(--cl-circle-transit)";
};

const getCircleColorFour = (props) => {
  return props.id === 4 ? "var(--cl-circle-4)" : "var(--cl-circle-transit)";
};

export const ProjectsContainer = styled(ScrollAnimation)`
  display: grid;
  grid-template-columns: repeat(10, 1fr);
  grid-template-rows: repeat(10, min-content);
  padding: 1rem;
`;

export const ProjecstTitle = styled.h2`
  grid-area: 1 / 1 / 2 / -1;
  font-family: var(--font-title);
  font-size: 4rem;
  font-weight: 300;
  padding: 1rem;
`;

export const ProjectsCircles = styled.div`
  grid-area: 2 / 6 / 3 / 9;
  display: grid;
  grid-template-columns: repeat(5, min-content);
  justify-content: end;
  grid-column-gap: 3rem;
  padding: 2rem;

  @media only screen and (max-width: 720px) {
    margin-top: 2rem;
    grid-area: 20 / 3 / 21 / 9;
    justify-content: center;
    grid-column-gap: 4rem;
  }

  @media only screen and (max-width: 425px) {
    grid-column-gap: 3rem;
  }
`;

export const ProjectsCircleZero = styled.div`
  ${circlesSharedStyle}
  background-color: ${getCircleColorZero};

  @media (hover: none) {
    &:hover {
      background-color: ${getCircleColorZero};
    }
  }
`;

export const ProjectsCircleOne = styled.div`
  ${circlesSharedStyle}
  background-color: ${getCircleColorOne};

  @media (hover: none) {
    &:hover {
      background-color: ${getCircleColorOne};
    }
  }
`;

export const ProjectsCircleTwo = styled.div`
  ${circlesSharedStyle}
  background-color: ${getCircleColorTwo};

  @media (hover: none) {
    &:hover {
      background-color: ${getCircleColorTwo};
    }
  }
`;

export const ProjectsCircleThree = styled.div`
  ${circlesSharedStyle}
  background-color: ${getCircleColorThree};

  @media (hover: none) {
    &:hover {
      background-color: ${getCircleColorThree};
    }
  }
`;

export const ProjectsCirclFour = styled.div`
  ${circlesSharedStyle}
  background-color: ${getCircleColorFour};

  @media (hover: none) {
    &:hover {
      background-color: ${getCircleColorFour};
    }
  }
`;

// ${projectsAnimation}
export const ProjectsImageWrapper = styled.div`
  position: relative;
  z-index: 5;

  grid-area: 2 / 2 / 8 / 5;

  box-shadow: 0 0.2rem 0.5rem var(--cl-square-shadow);
  animation: fadein 15s infinite;

  @media only screen and (max-width: 720px) {
    grid-area: 2 / 2 / 10 / 10;
  }

  ${fadeInProject}
`;

export const ProjectsImage = styled.img`
  width: 100%;
  height: 100%;
  display: block;
  object-fit: cover;
`;

export const ProjectsBox = styled.div`
  position: relative;
  grid-area: 3 / 3 / 10 / 10;
  display: grid;
  grid-template-columns: 3fr 7.5fr;

  background-color: var(--cl-grey);
  box-shadow: 0 0.2rem 1rem var(--cl-square-shadow);

  padding: 2rem;
  height: 40rem;

  @media only screen and (max-width: 768px) {
    height: 44rem;
  }

  @media only screen and (max-width: 720px) {
    grid-area: 7 / 1 / 20 / 11;
    z-index: 10;
  }

  @media only screen and (max-width: 310px) {
    height: 50rem;
  }

  @media only screen and (max-width: 254px) {
    height: 60rem;
  }
`;

// ${projectsAnimation}
export const ProjectsContent = styled.div`
  grid-column: 2 / 3;
  display: grid;
  grid-row-gap: 2rem;

  padding: 1rem;

  animation: fadein 15s infinite;

  @media only screen and (max-width: 720px) {
    grid-column: 1 / 3;
  }

  @media only screen and (max-width: 310px) {
    grid-row-gap: 1rem;
  }

  ${fadeInProject}
`;

export const ProjectsContentTitle = styled.h3`
  font-size: 2rem;
`;

export const ProjectsContentDesc = styled.div``;

export const ProjectsContentDescTitle = styled.span`
  font-weight: bold;
  color: ${getTitleColors};
`;

export const ProjectsContentDescText = styled.div``;

export const ProjectsContentBottom = styled.div`
  display: grid;
  grid-template-columns: 1fr min-content;
  align-items: center;

  @media only screen and (max-width: 254px) {
    grid-template-columns: 1fr;
  }
`;

export const ProjectsContentDate = styled.div``;

export const ProjectsContentLink = styled.a`
  background-color: ${getLinkColor};
  ${getThemeColorForLink}
  padding: 0.5rem 5rem;
  font-family: var(--font-title);
  text-decoration: none;
  text-align: center;

  transition: 0.3s background-color;

  &:hover {
    background-color: var(--cl-mainhover);
  }

  @media (hover: none) {
    &:hover {
      background-color: ${getLinkColor};
    }
  }
`;
