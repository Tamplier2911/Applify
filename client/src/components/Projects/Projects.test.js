import React from "react";
import toJson from "enzyme-to-json";

// projects data
import projectsData from "../../redux/projects/projectsConstants";

// component
import Projects from "./Projects.jsx";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<Projects />", () => {
  const initialStore = {
    theme: {
      currentTheme: "light",
    },
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
    projects: projectsData,
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<Projects store={store} />);
  });

  it("renders without an error", () => {
    const projects = findByTestAttr(wrapper, "projects");
    expect(projects).toHaveLength(1);
  });

  it("renders children - projects box", () => {
    const projectsBox = findByTestAttr(wrapper, "projects-box");
    expect(projectsBox).toHaveLength(1);
  });

  it("switches to first project on setCurrentProjectFirst", () => {
    const projectsSetFirst = findByTestAttr(wrapper, "projects-set-first");
    expect(projectsSetFirst).toHaveLength(1);

    const oldState = store.getState();
    expect(oldState.projects.currentProject).toEqual({});

    projectsSetFirst.simulate("click");
    const newState = store.getState();
    expect(newState.projects.currentProject).toEqual(
      projectsData.allProjects.eng[0]
    );
  });

  it("switches to second project on setCurrentProjectSecond", () => {
    const projectsSetSecond = findByTestAttr(wrapper, "projects-set-second");
    expect(projectsSetSecond).toHaveLength(1);

    const oldState = store.getState();
    expect(oldState.projects.currentProject).toEqual(
      projectsData.allProjects.eng[0]
    );

    projectsSetSecond.simulate("click");
    const newState = store.getState();
    expect(newState.projects.currentProject).toEqual(
      projectsData.allProjects.eng[1]
    );
  });

  it("switches to third project on setCurrentProjectThird", () => {
    const projectsSetThird = findByTestAttr(wrapper, "projects-set-third");
    expect(projectsSetThird).toHaveLength(1);

    const oldState = store.getState();
    expect(oldState.projects.currentProject).toEqual(
      projectsData.allProjects.eng[1]
    );

    projectsSetThird.simulate("click");
    const newState = store.getState();
    expect(newState.projects.currentProject).toEqual(
      projectsData.allProjects.eng[2]
    );
  });

  it("switches to fourth project on setCurrentProjectFourth", () => {
    const projectsSetFourth = findByTestAttr(wrapper, "projects-set-fourth");
    expect(projectsSetFourth).toHaveLength(1);

    const oldState = store.getState();
    expect(oldState.projects.currentProject).toEqual(
      projectsData.allProjects.eng[2]
    );

    projectsSetFourth.simulate("click");
    const newState = store.getState();
    expect(newState.projects.currentProject).toEqual(
      projectsData.allProjects.eng[3]
    );
  });

  it("switches to fifth project on setCurrentProjectFifth", () => {
    const projectsSetFifth = findByTestAttr(wrapper, "projects-set-fifth");
    expect(projectsSetFifth).toHaveLength(1);

    const oldState = store.getState();
    expect(oldState.projects.currentProject).toEqual(
      projectsData.allProjects.eng[3]
    );

    projectsSetFifth.simulate("click");

    const newState = store.getState();
    expect(newState.projects.currentProject).toEqual(
      projectsData.allProjects.eng[4]
    );
  });

  it("matches snapshot", () => {
    // renders projectsData.allProjects.eng[3]
    // current project after last simulation
    const projects = findByTestAttr(wrapper, "projects");
    expect(toJson(projects)).toMatchSnapshot();
  });
});
