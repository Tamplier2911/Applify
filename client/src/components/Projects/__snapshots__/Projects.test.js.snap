// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> matches snapshot 1`] = `
<Styled(ScrollAnimation)
  animateIn="bounceInRight"
  animateOnce={true}
  data-test="projects"
>
  <styled.h2>
    Recent projects:
  </styled.h2>
  <styled.div>
    <styled.div
      data-test="projects-set-first"
      id={4}
      onClick={[Function]}
    />
    <styled.div
      data-test="projects-set-second"
      id={4}
      onClick={[Function]}
    />
    <styled.div
      data-test="projects-set-third"
      id={4}
      onClick={[Function]}
    />
    <styled.div
      data-test="projects-set-fourth"
      id={4}
      onClick={[Function]}
    />
    <styled.div
      data-test="projects-set-fifth"
      id={4}
      onClick={[Function]}
    />
  </styled.div>
  <styled.div
    load={false}
  >
    <styled.img
      alt="Project presentation slide."
      src="clarify-min.jpg"
    />
  </styled.div>
  <styled.div
    data-test="projects-box"
  >
    <styled.div
      load={false}
    >
      <styled.h3>
        Clarify
      </styled.h3>
      <styled.div>
        <styled.span
          color="#DD4F4F"
        >
          Description:
        </styled.span>
        <styled.div>
          Clarify - is an email service, that helps you to collect feedback from people in order to improve their experience.
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.span
          color="#DD4F4F"
        >
          Technology Stack:
        </styled.span>
        <styled.div>
          Front-end of the app covered by React alongside with Redux and Redux-Sagas. Back-end handled with Express server, data persistance achieved with MongoDB and Mongoose
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div>
          August 12, 2019
        </styled.div>
        <styled.a
          color="#DD4F4F"
          href="https://clarify-s.herokuapp.com/"
          rel="noopener noreferrer"
          target="_blank"
          theme="light"
        >
          Visit
        </styled.a>
      </styled.div>
    </styled.div>
  </styled.div>
</Styled(ScrollAnimation)>
`;
